pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -127,22 +130,39 @@ def setup_mip_solver(solve_data, config, regularization_problem):
     if regularization_problem:
         mainopt = SolverFactory(config.mip_regularization_solver)
     else:
-        mainopt = SolverFactory(config.mip_solver)
+        if config.mip_solver == 'gurobi_persistent' and config.single_tree:
+            @SolverFactory.register('gurobi_persistent4mindtpy', doc='Persistent python interface to Gurobi')
+            class GurobiPersistent4MindtPy(GurobiPersistent):
```

Comment:
Is this standard naming for classes in Pyomo? With numbers and everything?

Reply:
I don't know. Maybe the developer group can give us some suggestions.